code/src/cljs/sixsq/nuvla/ui/notifications/views.cljs

```diff
@@ -0,0 +1,930 @@
+(ns sixsq.nuvla.ui.notifications.views
+  (:require
+    [clojure.string :as str]
+    [re-frame.core :refer [dispatch dispatch-sync subscribe]]
+    [reagent.core :as r]
+    [sixsq.nuvla.ui.history.views :as history]
+    [sixsq.nuvla.ui.notifications.events :as events]
+    [sixsq.nuvla.ui.notifications.spec :as spec]
+    [sixsq.nuvla.ui.notifications.subs :as subs]
+    [sixsq.nuvla.ui.i18n.subs :as i18n-subs]
+    [sixsq.nuvla.ui.main.components :as main-components]
+    [sixsq.nuvla.ui.panel :as panel]
+    [sixsq.nuvla.ui.utils.form-fields :as ff]
+    [sixsq.nuvla.ui.utils.general :as general-utils]
+    [sixsq.nuvla.ui.utils.semantic-ui :as ui]
+    [sixsq.nuvla.ui.utils.semantic-ui-extensions :as uix]
+    [sixsq.nuvla.ui.utils.style :as style]
+    [sixsq.nuvla.ui.utils.ui-callback :as ui-callback]
+    [sixsq.nuvla.ui.utils.validation :as utils-validation]
+    [taoensso.timbre :as timbre]))
+
+
+(defn save-callback-notification-subscription-config
+  []
+  (dispatch-sync [::events/set-validate-form? true])
+  (dispatch-sync [::events/validate-notification-subscription-config-form])
+  (let [form-valid? (get @re-frame.db/app-db ::spec/form-valid?)]
+    (when form-valid?
+      (do
```

Comment:
`when` is able to have multiple statement in it's body. The do is useless

Reply:
changed